RUM-12611 Update readme for Apollo 2.0.0 compatibility

barboraplasovska opened the pull request:
### What and why?

Update the readme to mention the steps to setup Apollo Interceptor with Apollo 2.0.0+.

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes
- [ ] Add Objective-C interface for public APIs (see our [guidelines](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/3157787243/RFC+-+Modular+Objective-C+Interface#Recommended-solution) (internal) and run `make api-surface`)

barboraplasovska committed:
RUM-12611 Update readme for Apollo 2.0.0 compatibility

joepeeples commented:
Opened [DOCS-12715](https://datadoghq.atlassian.net/browse/DOCS-12715) to assign a Docs writer and follow up with editorial review.

[DOCS-12715]: https://datadoghq.atlassian.net/browse/DOCS-12715?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

blimmer committed:
fix: change id property to var for Apollo iOS 1.x compatibility

Changed the id property from let to var to properly conform to the
ApolloInterceptor protocol in Apollo iOS 1.x versions. This ensures
backward compatibility with older Apollo versions while maintaining
forward compatibility with newer ones.

Fixes #16

barboraplasovska committed:
Merge pull request #18 from homebound-team/fix-issue-16

fix: change id property to var for Apollo iOS 1.x compatibility

barboraplasovska committed:
Fix: Remove ssh secret

barboraplasovska committed:
Merge pull request #20 from DataDog/bplasovska/remove-ssh-secret

Fix: Remove ssh secret

rtrieu commented on .gitlab-ci.yml:
```diff
@@ -2,11 +2,21 @@ stages:
+  # Prefilled variables for running a pipeline manually:
+  # Ref.: https://docs.gitlab.com/ee/ci/pipelines/index.html#prefill-variables-in-manual-pipelines
+  RELEASE_GIT_TAG:
+    description: "The Git tag for the release pipeline. If set, release pipeline will be triggered for the given tag."
```
```suggestion
    description: "The Git tag for the release pipeline. If set, release pipeline is triggered for the given tag."
```

rtrieu commented on .gitlab-ci.yml:
```diff
@@ -2,11 +2,21 @@ stages:
+    description: "The Git tag for the release pipeline. If set, release pipeline will be triggered for the given tag."
+  RELEASE_DRY_RUN:
+    value: "1"
+    description: "Controls the dry run mode for the release pipeline. If set to '1', the pipeline will execute all steps but will not publish artifacts. If set to '0', the pipeline will run fully."
```
```suggestion
    description: "Controls the dry run mode for the release pipeline. If set to '1', the pipeline executes all steps but does not publish artifacts. If set to '0', the pipeline will run fully."
```

rtrieu commented on README.md:
```diff
@@ -36,7 +50,62 @@ class CustomInterceptorProvider: DefaultInterceptorProvider {
-This automatically adds Datadog headers to your GraphQL requests, enabling them to be tracked by Datadog. Note that while `query` and `mutation` operations are tracked, `subscription` operations are not.
+### For Apollo iOS 2.0+
+
+1. Set up RUM monitoring with the [Datadog iOS RUM SDK](https://docs.datadoghq.com/real_user_monitoring/ios/):
```
```suggestion
1. Set up RUM monitoring with the [Datadog iOS SDK](https://docs.datadoghq.com/real_user_monitoring/ios/):
```

rtrieu commented on SmokeTests/spm/App/AppDelegate.swift:
```diff
@@ -0,0 +1,23 @@
+/*
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
```
```suggestion
* Unless explicitly stated otherwise, all files in this repository are licensed under the Apache License Version 2.0.
```

rtrieu commented on SmokeTests/spm/App/SceneDelegate.swift:
```diff
@@ -0,0 +1,36 @@
+/*
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
```
```suggestion
* Unless explicitly stated otherwise, all files in this repository are licensed under the Apache License Version 2.0.
```

rtrieu commented on SmokeTests/spm/App/ViewController.swift:
```diff
@@ -0,0 +1,34 @@
+/*
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
```
```suggestion
* Unless explicitly stated otherwise, all files in this repository are licensed under the Apache License Version 2.0.
```

rtrieu commented on SmokeTests/spm/AppTests/AppTests.swift:
```diff
@@ -0,0 +1,17 @@
+/*
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
```
```suggestion
* Unless explicitly stated otherwise, all files in this repository are licensed under the Apache License Version 2.0.
```

rtrieu commented on SmokeTests/spm/AppUITests/AppUITests.swift:
```diff
@@ -0,0 +1,18 @@
+/*
+* Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
```
```suggestion
* Unless explicitly stated otherwise, all files in this repository are licensed under the Apache License Version 2.0.
```

rtrieu commented on README.md:
```diff
@@ -2,25 +2,39 @@
 1. Go to File → Add Package Dependencies
 2. Enter the repository URL: `https://github.com/DataDog/dd-sdk-ios-apollo-interceptor`
-3. Select the latest version
+3. Select version 1.x.x for Apollo iOS 1.0+ or version 2.x.x for Apollo iOS 2.0+
```
```suggestion
3. Select the package version that matches your Apollo major version (choose 1.x.x for Apollo iOS 1.0+ or 2.x.x for Apollo iOS 2.0+).
```

rtrieu reviewed (approved):
left you some minor suggestions for your review!

barboraplasovska committed:
Apply CR suggestions
